CMakeLists.txt

```diff
@@ -231,12 +232,21 @@ endif ()
 # library.
 add_definitions(-DBOOST_TEST_DYN_LINK)
 
-# Detect OpenMP support in a compiler. If the compiler supports OpenMP, the
-# flags to compile with OpenMP are returned and added.
+# Detect OpenMP support in a compiler. If the compiler supports OpenMP, flags 
+# to compile with OpenMP are returned and added and the HAS_OPENMP is set to 1. 
+# This way we can skip calls to functions defined in omp.h with code like:
+# if (HAS_OPENMP == 1) { openMP code here }
+# If OpenMP is found, define HAS_OPENMP to be 1. Otherwise define it to be 0.
 find_package(OpenMP)
 if (OPENMP_FOUND)
+  add_definitions(-DHAS_OPENMP)
+  set(HAS_OPENMP "1")
```

Comment:
I'm not sure if the CMake variable `HAS_OPENMP` is necessary; it doesn't look like it's used anywhere.  Maybe I overlooked something?